lib/pymedphys/_dicom/delivery/core.py

```diff
@@ -37,15 +41,106 @@ def _pretty_print(string):
     return "\n\n".join(wrapped)
 
 
+def _check_for_supported_collimation_device(
+    beam_limiting_device_sequence: "pydicom.Sequence",
+):
+    """Validate whether or not the beam limiting devices in use are
+    supported for use by ``pymedphys.Delivery.from_dicom``.
+
+    Parameters
+    ----------
+    beam_limiting_device_sequence
+        The ``BeamLimitingDeviceSequence`` from within a single item
+        within the ``BeamSequence`` of an RT Plan DICOM dataset.
+
+    Raises
+    ------
+    ValueError
+        If the device types are not contained within the supported
+        configrations.
+
+    """
+    rt_beam_limiting_device_types = {
+        item.RTBeamLimitingDeviceType for item in beam_limiting_device_sequence
+    }
+
+    supported_configurations = [{"MLCX", "ASYMY"}]
+
+    if not rt_beam_limiting_device_types in supported_configurations:
+        raise ValueError(
+            _pretty_print(
+                """\
+                    Currently only DICOM files where the beam
+                    limiting devices consist of one of the following
+                    combinations are supported:
+
+                    * {supported_configurations}
+
+                    The provided RT Plan DICOM file has the
+                    following:
+
+                        {rt_beam_limiting_device_types}
+
+                    This is not yet supported.
+                    This is due to a range of assumptions being made
+                    internally that assume a single jaw system.
+                    There are some cases where this restriction is
+                    too tight. Currently however there is not enough
+                    testing data to appropriately implement these
+                    cases.
+                    If you would like to have your device supported
+                    please consider uploading anonymised DICOM files
+                    and their TRF counterparts to the following
+                    issue
+                    <https://github.com/pymedphys/pymedphys/issues/1142>.
+
+                """
+            ).format(
+                supported_configurations="\n* ".join(
+                    [str(item) for item in supported_configurations]
+                ),
+                rt_beam_limiting_device_types=rt_beam_limiting_device_types,
+            )
+        )
+
+
 class DeliveryDicom(DeliveryBase):
     @classmethod
-    def from_dicom(cls, dicom_dataset, fraction_number=None):
+    def from_dicom(cls, rtplan: dicom_path_or_dataset, fraction_number=None):
+        """Create a ``pymedphys.Delivery`` object from an RT Plan DICOM
+        dataset.
+
+        Parameters
+        ----------
+        rtplan : pydicom.Dataset or pathlib.Path
+            An RT Plan DICOM dataset, or the filepath to such a dataset.
+        fraction_number : 'all' or int, optional
```

Comment:
So it's actually FractionGroupNumber, and not fraction number...
I suggest renaming the variable.